atom/browser/lib/guest-view-manager.coffee

```diff
@@ -38,12 +39,23 @@ moveLastToFirst = (list) ->
 getNextInstanceId = (webContents) ->
   ++nextInstanceId
 
+# Generate URL encoded partition id.
+getPartitionId = (partition) ->
+  persist = partition.startsWith('persist:')
+  # Guest site url will be chrome-guest://fake-host/{persist}?{partitionId}
+  partitionId = "chrome-guest://fake-host/"
+  if partition
+    partitionId += if persist then 'persist?' else '?'
+    partitionId += crypto.createHash('sha256').update(partition).digest('hex')
```

Comment:
> This way it is easy to find out what's under the data dir, we would also have to notice users about valid partition names though

have changed to encode the partitionId before creating the URI, that would fix the problem ?